WordPress/Classes/Networking/PostServiceRemoteXMLRPC.m

```diff
@@ -164,6 +169,54 @@ - (void)deletePost:(RemotePost *)post
     }
 }
 
+- (void)trashPost:(RemotePost *)post
+          forBlog:(Blog *)blog
+          success:(void (^)(RemotePost *))success
+          failure:(void (^)(NSError *))failure
+{
+    NSParameterAssert([post.postID longLongValue] > 0);
+    NSNumber *postID = post.postID;
+    if ([postID longLongValue] > 0) {
+        NSArray *parameters = [blog getXMLRPCArgsWithExtra:postID];
+
+        WPXMLRPCRequest *deletePostRequest = [self.api XMLRPCRequestWithMethod:@"wp.deletePost" parameters:parameters];
```

Comment:
I'll rephrase: let's break to 80 chars the lines that can be broken to 80 chars, and bring down to what we can the ones that can't.  I'm mentioning it here only but this is something I'm seeing throughout the PR that could be improved a bit to make the code easier to read.  For example:

```objective-c
        WPXMLRPCRequest *deletePostRequest = [self.api XMLRPCRequestWithMethod:@"wp.deletePost" parameters:parameters];
        WPXMLRPCRequestOperation *delOperation = [self.api XMLRPCRequestOperationWithRequest:deletePostRequest success:nil failure:nil];

        WPXMLRPCRequest *getPostRequest = [self.api XMLRPCRequestWithMethod:@"wp.getPost" parameters:parameters];
        WPXMLRPCRequestOperation *getOperation = [self.api XMLRPCRequestOperationWithRequest:getPostRequest success:^(AFHTTPRequestOperation *operation, id responseObject) {
            if (success) {
                // The post was trashed but not yet deleted.
                RemotePost *post = [self remotePostFromXMLRPCDictionary:responseObject];
                success(post);
            }
        } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
            if (error.code == 404) {
                // The post was deleted.
                if (success) {
                    success(post);
                }
            }
        }];
```

...becomes...

```objective-c
        WPXMLRPCRequest *deletePostRequest = [self.api XMLRPCRequestWithMethod:@"wp.deletePost"
                                                                    parameters:parameters];
        WPXMLRPCRequestOperation *delOperation = [self.api XMLRPCRequestOperationWithRequest:deletePostRequest
                                                                                     success:nil
                                                                                     failure:nil];

        WPXMLRPCRequest *getPostRequest = [self.api XMLRPCRequestWithMethod:@"wp.getPost"
                                                                 parameters:parameters];
        WPXMLRPCRequestOperation *getOperation = [self.api XMLRPCRequestOperationWithRequest:getPostRequest
                                                                                     success:^(AFHTTPRequestOperation *operation, id responseObject) {
            if (success) {
                // The post was trashed but not yet deleted.
                RemotePost *post = [self remotePostFromXMLRPCDictionary:responseObject];
                success(post);
            }
        } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
            if (error.code == 404) {
                // The post was deleted.
                if (success) {
                    success(post);
                }
            }
        }];
```